pulp_ansible/app/tasks/collections.py

```diff
@@ -1154,13 +1154,15 @@ def _pre_save(self, batch):
                 d_content.content.namespace = namespace
                 if d_content.d_artifacts:
                     da = d_content.d_artifacts[0]
-                    # Check to see if avatar failed to download, update metadata if so
-                    if da.deferred_download:
-                        d_content.d_artifacts = None
-                        d_content.content.avatar_sha256 = None
-                    # Check to see if upstream didn't have avatar_sha256 set
-                    elif d_content.content.avatar_sha256 is None:
-                        d_content.content.avatar_sha256 = da.artifact.sha256
+                    if False:
+                        # This code invalidates the digest.
+                        # Check to see if avatar failed to download, update metadata if so
+                        if da.deferred_download:
+                            d_content.d_artifacts = None
+                            d_content.content.avatar_sha256 = None
+                        # Check to see if upstream didn't have avatar_sha256 set
+                        elif d_content.content.avatar_sha256 is None:
+                            d_content.content.avatar_sha256 = da.artifact.sha256
```

Comment:
```suggestion
                      # Check to see if avatar failed to download, update metadata if so
                      if da.deferred_download:
                          d_content.d_artifacts = None
                          d_content.content.avatar_sha256 = None
                          d_content.content.metadata_sha256 = None
```
How about this? If `da.deferred_download` is true that means the avatar failed to download. We null the `avatar_sha256` and `metadata_sha256` (<-this gets recalculated on save), that way on next sync we will try download the image again.

Reply:
Interesting idea. Let's try.